File: desktop operating systems/ui/WDT/src/app/Configuration/WorkingWithDataBaseConfiguration.java
package app.Configuration;

import app.Common.Options;
import com.WDTComponents.DataBase.IWorkingWithDataBase;

import java.sql.*;
import java.util.ArrayList;

public class WorkingWithDataBaseConfiguration implements IWorkingWithDataBase {

    private Connection connection;
    private Statement statement;

    @Override
    public boolean createDataBase() {
        try {
            connection = DriverManager.getConnection(
                    "jdbc:sqlite:"+ Options.INSTANCE.getCOMMON_DIRECTORY() + "/" +
                            Options.APPLICATION_TITLE + "_Database.db"
            );
            statement = connection.createStatement();
            return true;
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return false;
    }

    @Override
    public synchronized boolean executeQuery(String SQLQuery) {
        try {
            statement.executeUpdate(SQLQuery);
            return true;
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return false;
    }

    @Override
    public synchronized ArrayList<String[]> executeRowQuery(String SQLQuery) {
        ArrayList<String[]> arrayList = new ArrayList<>();
        try {
            ResultSet resultSet = statement.executeQuery(SQLQuery);
            while (resultSet.next()) {
                String[] tempArrayString = new String[resultSet.getMetaData().getColumnCount()];
                for (int i = 1; i <= resultSet.getMetaData().getColumnCount(); i++)
                    tempArrayString[i - 1] = resultSet.getString(i);
                arrayList.add(tempArrayString);
            }
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return arrayList;
    }

    @Override
    public void closeConnection() {
        try {
            connection.close();
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }

}

File: android operating system/WirelessDataTransmission_WDT/app/src/main/java/ua/edu/onaft/wirelessdatatransmission_wdt/Configuration/WorkingWithDataBaseConfiguration.kt
package ua.edu.onaft.wirelessdatatransmission_wdt.Configuration

import android.annotation.SuppressLint
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.WDTComponents.DataBase.FeedReaderContract
import com.WDTComponents.DataBase.IWorkingWithDataBase
import ua.edu.onaft.wirelessdatatransmission_wdt.Common.StaticState
import java.util.*
import kotlin.collections.ArrayList

class WorkingWithDataBaseConfiguration: IWorkingWithDataBase, SQLiteOpenHelper(StaticState.activity.applicationContext, DATABASE_NAME, null, DATABASE_VERSION) {

    override fun closeConnection() {}

    override fun createDataBase(): Boolean { return true }

    override fun executeQuery(SQLQuery: String): Boolean = try {
        this.writableDatabase.execSQL(SQLQuery)
        true
    } catch (E: Exception) { false }

    @SuppressLint("Recycle")
    override fun executeRowQuery(SQLQuery: String): ArrayList<Array<String>> {
        val arrayList: ArrayList<Array<String>> = ArrayList()
        val resultSet: Cursor = this.writableDatabase.rawQuery(SQLQuery, null)
        while (resultSet.moveToNext()) {
            val tempArrayString: Array<String?> = arrayOfNulls(resultSet.columnCount)
            for (i in 1 .. resultSet.columnCount)
                tempArrayString[i - 1] = resultSet.getString(i-1)
            arrayList.add(tempArrayString as Array<String>)
        }
        return arrayList
    }

    override fun onCreate(db: SQLiteDatabase?) {
        db?.execSQL(FeedReaderContract.SQL_CREATE_DEVICE)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        db?.execSQL(FeedReaderContract.SQL_DROP_DEVICE)
        onCreate(db)
    }

    companion object {
        const val DATABASE_VERSION = 1
        const val DATABASE_NAME = "WirelessDataTransmission (WDTFX)_Database.db"
    }

}